public/app/features/dashboard/components/GenAI/GenAIPanelDescriptionButton.tsx

```diff
@@ -24,7 +24,9 @@ const DESCRIPTION_GENERATION_STANDARD_PROMPT =
   `The description should be, at most, ${PANEL_DESCRIPTION_CHAR_LIMIT} characters.`;
 
 export const GenAIPanelDescriptionButton = ({ onGenerate, panel }: GenAIPanelDescriptionButtonProps) => {
```

Comment:
Probably we can generate the messages when clicking, passing `messages` as callback, as we did for the save changes summary. This ensure this is called only once.